system/xmllib/libxml_xpathprocessor.cpp

```diff
@@ -978,6 +978,35 @@ class CLibXpathContext : public CInterfaceOf<IXpathContext>, implements IVariabl
         return xml;
     }
 
+    virtual bool prepareTrace(ICompiledXpath* select, StringBuffer& content, bool& isValue)
```

Comment:
This shares an awful lot of code with the trace function below.  Can trace call prepareTrace?  I haven't looked yet, but I'm wondering if/where the trace function is still being used.

Reply:
Trace should not, in my opinion, be called. I would like to remove it, but wasn't sure how much backward compatibility would be a concern. If it remains, it should be changed to use prepareTrace.

Reply:
If trace remains, it will not mask so keeping it is a risk.

I'm also unsure about the name prepareTrace. It makes sense in the sense that it prepares the content that will be traced. I just don't know if trace output is the only reason we will ever need this functionality, or if a more general name would be better.

Reply:
As we discussed offline, I agree we should get rid of the old function, and name/design prepareTrace more generically since it could be useful for serializing content beyond tracing.